### ISSUE TITLE
Static linking against ThirdParty/HSL fails

### ISSUE THREAD

**Issue by ianbeyst** [coin-or/Ipopt#373]
When trying to build a static ipopt library with HSL included, using the command
`bash coinbrew build Ipopt --skip='ThirdParty/ASL ThirdParty/Mumps' --enable-static --disable-shared`, the compilation fails. [Here is the full output](https://github.com/coin-or/Ipopt/files/4503044/output.txt). I first fetched the code, inserted the HSL solver code as obtained from http://www.hsl.rl.ac.uk/ipopt/ to the `ThirdParty/HSL/coinhsl` directory and it is compiled and installed without problem.

The cause seems to be that the linear solver loader is disabled when passing `--disable-shared`. Then, in line 17 of `Ipopt/src/Algorithm/LinearSolvers/IpMa27TSolverInterface.cpp` the `#if` evaluates to true since `COIN_HAS_HSL` is true. The references inside this `#if` block to `ma27id` etcetera are only defined in the `HSLLoader.h` file however, as far as I can tell. Since this header is only included when compiling in the linear solver loader which is not the case, these references are undefined.

**Comment by svigerske** [comment]
`HSL_FUNC` should be defined in config.h and Ipopt's configure should figure out what to define this to.
That probably has failed. So far that is only a warning in configure, but I should change this to an error.

With the coinbrew-default of suppressing all useful output, it is difficult to see what has happened. Can you rerun with verbosity increased (`--verbosity 4`) and if there is such a warning, look into the config.log why this has happened. ~~I guess it is due to missing references to some gfortran symbols, which only happens if ThirdParty/HSL has been build as static lib.~~ (`FLIBS` should make it into coinhsl.pc)

**Comment by ianbeyst** [comment]
I don't get any warnings not already printed in the output above. The warning in the earlier output about the unknown mangling scheme is related to HSL though:
`checking HSL name mangling scheme... unknown` immediately followed by
`configure: WARNING: Unsupported or unknown name-mangling scheme: unknown`.
Besides that I don't notice anything unusual in the configure output, but maybe I'm missing something.
[Here is the output with `--verbosity 4`.](https://github.com/coin-or/Ipopt/files/4503769/output.txt)

**Comment by svigerske** [comment]
~~How does `/home/io/tmp/dist/lib/pkgconfig/coinhsl.pc` look like?~~

~~And does `/home/io/tmp/build/Ipopt/master/config.log` say why it had problems to link with the HSL lib? Search for `ma28ad` and what follows.~~

I can reproduce and see what is going on.

**Comment by svigerske** [comment]
It should be fixed if you use branch `stable/3.13` (which will be merged into master sometime soon).

```
coinbrew fetch Ipopt:stable/3.13
<place HSL source into ThirdParty/HSL>
coinbrew build Ipopt:stable/3.13 --skip='ThirdParty/ASL ThirdParty/Mumps' --disable-shared --no-prompt --verbosity=4
```
